### DISCUSSION TITLE
how to review "auto-cooled" GCode

### DISCUSSION THREAD

**Discussion by raemin** [arkypita/LaserGRBL#2140] [General]
sorry for the so-so title...

I am using QCad with laserGrbl scripts so as to produce GCode that I the process through LaserGRBL. 

Is it possible to review what was actually sent to the GRBL board?

I am trying to review how the cooling sequences are embeded in the code initially produced by QCAD.

**Comment by arkypita** [comment]
Any gcode file is a text file, just open it with notepad.

**Reply by raemin** [reply]
That's precisely wbat I want to do, open in notepad the code that was modified by LaserGRBL, i. e my original QCAD GRBL with the Cooling instructions that LaserGRBL adds every 10minutes.

**Reply by raemin** [reply]
Is this something that somehow appears in the log?

Thanks

**Reply by arkypita** [reply]
LaserGRBL does not add gcode to your code for auto-cooling.

LaserGRBL use grbl firmware features (immediate commands as feed hold and resume) to command the controller immediate stop and resume.

This commands are firmware specific, and are not part of gcode stream.

https://github.com/gnea/grbl/wiki/Grbl-v1.1-Interface#real-time-control-commands

**Reply by raemin** [reply]
The link  makes it perfectly clear.

Managing these commands in a real-time process has some clear advantages, unfortunately I won't be able to tweak this on the QCAD side. No big deal though.

Thanks
